libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -1218,23 +1373,29 @@ ManagedQuery::_extend_and_evolve_schema_with_details(
     ArrowSchema* value_schema,
     ArrowArray* value_array,
     std::string column_name,
+    bool deduplicate,
     Enumeration enmr,
     ArraySchemaEvolution& se) {
-    // We need to check if we are writing any new enumeration values. If so,
-    // extend and evolve the schema. If not, just set the write buffers to the
-    // dictionary's indexes as-is
+    if (value_array->n_buffers != 2) {
+        throw std::invalid_argument(fmt::format(
+            "[ManagedQuery] _extend_and_evolve_schema_with_details non-string: "
+            "expected n_buffers == 2; got {}",
+            value_array->n_buffers));
+    }
 
     // Get all the enumeration values in the passed-in column
     std::vector<ValueType> enum_values_in_write;
     uint64_t num_elems = value_array->length;
     if (strcmp(value_schema->format, "b") == 0) {
```

Comment:
why is the bool-specific code in the general purpose template? 

related to this: the caller treats bools as if they are the same as uint8, and then catches the problem here.  Seems like a very confusing way to handle the generic method, which is otherwise parameterized by arrow array type.

Recommend that you either have a specialization for the Arrow bool_ type, _or_ perform this explicit cast to uint8 before calling this method

Reply:
I stand with the current factoring. It's just an if-block, and I find extending the profusion of helper methods more bulky than clarifying.